akka-persistence-testkit/src/main/scala/akka/persistence/testkit/javadsl/SnapshotTestKit.scala

```diff
@@ -0,0 +1,305 @@
+   */
+  def expectPersistedInOrder[A](persistenceId: String, msgs: JList[A], max: Duration): JList[A] =
+    scalaTestkit.expectPersistedInOrder(persistenceId, msgs, max.asScala).asJava
+
+  /**
+   * Check that `msgs` events have been persisted in the storage regardless of order.
+   */
+  def expectPersistedInAnyOrder[A](persistenceId: String, msgs: JList[A]): JList[A] =
+    scalaTestkit.expectPersistedInAnyOrder(persistenceId, msgs).asJava
+
+  /**
+   * Check for `max` time that `msgs` events have been persisted in the storage regardless of order.
+   */
+  def expectPersistedInAnyOrder[A](persistenceId: String, msgs: JList[A], max: Duration): JList[A] =
+    scalaTestkit.expectPersistedInAnyOrder(persistenceId, msgs, max.asScala).asJava
+
+  /**
+   * Persist `elems` snapshots with metadata into storage in order.
+   */
+  def persistForRecovery(persistenceId: String, elems: JList[Pair[SnapshotMeta, Any]]): Unit =
+    scalaTestkit.persistForRecovery(persistenceId, elems.map(_.toScala))
+
+  /**
+   * Retrieve all snapshots and their metadata saved in storage by persistence id.
+   */
+  def persistedInStorage(persistenceId: String): JList[Pair[SnapshotMeta, Any]] =
+    scalaTestkit.persistedInStorage(persistenceId).map(p => Pair(p._1, p._2)).asJava
+
+  /**
+   * Clear all data from storage.
+   */
+  def clearAll(): Unit = scalaTestkit.clearAll()
+
+  /**
+   * Clear all data from storage for particular persistence id.
+   */
+  def clearByPersistenceId(persistenceId: String): Unit = scalaTestkit.clearByPersistenceId(persistenceId)
+
+  /**
+   * Fail `n` following journal events depending on the condition `cond`.
+   * Failure triggers, when `cond` returns true, .
+   * Fails events with default `ExpectedFailure` exception.
+   */
+  def failNextNOpsCond(cond: jf.BiFunction[String, SnapshotOperation, Boolean], n: Int): Unit =
+    failNextNOpsCond(cond, n, ExpectedFailure)
+
+  /**
+   * Fail `n` following journal events depending on the condition `cond`.
+   * Failure triggers, when `cond` returns true, .
+   * Fails events with the `cause` exception.
+   */
+  def failNextNOpsCond(cond: jf.BiFunction[String, SnapshotOperation, Boolean], n: Int, cause: Throwable): Unit =
+    scalaTestkit.failNextNOpsCond(cond, n, cause)
+
+  /**
+   * Fail n following journal events regardless of their type.
+   * Fails events with default `ExpectedFailure` exception.
+   */
+  def failNextNOps(n: Int): Unit =
+    failNextNOps(n, ExpectedFailure)
+
+  /**
+   * Fail `n` following journal events depending on the condition `cond`.
+   * Failure triggers, when `cond` returns true, .
+   * Fails events with the `cause` exception.
+   */
+  def failNextNOps(n: Int, cause: Throwable): Unit = scalaTestkit.failNextNOps(n, cause)
+
+  /**
+   * Set new processing policy for journal events.
+   * NOTE! Overrides previously invoked `failNext...` or `rejectNext...`
+   */
+  def withPolicy(policy: SnapshotStorage.SnapshotPolicies.PolicyType): this.type = {
```

Comment:
`this.type` isn't great for Java. not sure if it will work at all, but this can be `: SnapshotTestKit` instead